testing/integration/package_version_test.go

```diff
@@ -102,23 +103,40 @@ func TestComponentBuildHashInDiagnostics(t *testing.T) {
 			return false
 		}
 
+		if client.State(status.State) != client.Healthy {
+			stateBuff.WriteString(fmt.Sprintf(
+				"agent isn't healthy: %s-%s",
+				client.State(status.State), status.Message))
+			return false
+		}
+
+		if len(status.Components) == 0 {
```

Comment:
Please add a comment above this to explain why this check is required being it really fixed the flaky issue in the test.